test/export/test_safeguard.py

```diff
@@ -0,0 +1,60 @@
+# Owner(s): ["module: dynamo"]
+import torch
+from torch.export import export
+from torch.testing._internal.common_utils import run_tests, TestCase
+
+
+class TestSafeguard(TestCase):
+    def test_grad_mode_unsupported(self):
+        def f1(a):
+            with torch.no_grad():
+                b = a + a
+            return b
+
+        def f2(a):
+            with torch.enable_grad():
+                b = a + a
+            return b
+
+        def f3(a):
+            with torch.set_grad_enabled(False):
+                b = a + a
+            return b
+
+        def f4(a):
+            with torch.set_grad_enabled(True):
```

Comment:
I learned from Brian that this is unsafe to globally manipulate state like this. Can you wrap the entire test function in a context manager that set the old values back after the test is complete?

Reply:
I think it will revert to the previous status when jumping out of the context. Is what you expect?

Reply:
Oh oops sorry missed the context manager part. No need to change